src/commands/importExportEntities/entities/utils/typeElements.ts

```diff
@@ -0,0 +1,260 @@
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+      });
+    }
+    case "rich_text": {
+      const typedElement = element as ContentTypeElements.IRichTextElement;
+      return params.builder.richTextElement({
+        ...typedElement,
+        type: "rich_text",
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+        allowed_content_types: undefined, // will be set later (when types are imported) to be able to map the ids
+        allowed_item_link_types: undefined, // will be set later (when types are imported) to be able to map the ids
+      })
+    }
+    case "snippet": {
+      if (!("snippetElement" in params.builder)) {
+        throw new Error(`Type snippet with codename "${params.typeOrSnippetCodename}" has an element (codename: "${element.codename}") of type snippet which is not allowed.`);
+      }
+
+      const typedElement = element as unknown as ContentTypeElements.ISnippetElement;
+      return params.builder.snippetElement({
+        ...typedElement,
+        type: "snippet",
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+        snippet: { id: params.context.contentTypeSnippetIdsWithElementsByOldIds.get(typedElement.snippet.id ?? "")?.selfId },
+      });
+    }
+    case "subpages": {
+      const typedElement = element as ContentTypeElements.ISubpagesElement;
+      return params.builder.subpagesElement({
+        ...typedElement,
+        type: "subpages",
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+        allowed_content_types: undefined, // will be set later (when types are imported) to be able to map the ids
+        default: undefined, // will be set later (when items are imported) to be able to map the ids
+      } as ContentTypeElements.ISubpagesElementData);
+    }
+    case "taxonomy": {
+      const typedElement = element as ContentTypeElements.ITaxonomyElement;
+      return params.builder.taxonomyElement({
+        ...typedElement,
+        type: "taxonomy",
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+        taxonomy_group: { id: params.context.taxonomyGroupIdsByOldIds.get(typedElement.taxonomy_group.id ?? "") },
+        default: typedElement.default ? { global: { value: typedElement.default.global.value.map(ref => ({ id: params.context.taxonomyTermIdsByOldIds.get(ref.id ?? "") })) } } : undefined,
+      });
+    }
+    case "text": {
+      const typedElement = element as ContentTypeElements.ITextElement;
+      return params.builder.textElement({
+        ...typedElement,
+        type: "text",
+        external_id: typedElement.external_id ?? fallbackExternalId,
+        content_group,
+      });
+    }
+    case "url_slug": {
+      const typedElement = element as ContentTypeElements.IUrlSlugElement;
+      if (!("urlSlugElement" in params.builder)) {
```

Comment:
This is kind of hard to follow at first, would not this condition be a good case for typeGuard? also with snippet